### PR TITLE
Fix #7118: Adding Filter by flag icon to card browser.

### LINKED ISSUE #7118: Browser: Enable Filtering on Flags
Users don't know that they can type: `flag:1` etc... this needs a visual item in the Card Browser Action bar - similar to the item in the Reviewer

### PR DESCRIPTION
**Pull request by Volfee**
## Purpose / Description
Users don't know that they can type: flag:1 etc... this needs a visual item in the Card Browser Action bar - similar to the item in the Reviewer.

## Fixes
Fixes #7118

## Approach
1. Flag icon is added to card browser menu with ifRoom option. After that change it will replace "Add card button" on most mobile phones, since there is not enough room for both and search icon.
2. When flag is selected, searchTerms are updated to have `flag:N`.
3. If there was already search term looked for by user, selecting flag will prepend `flag:N` to search terms.
4. If user has selected a flag and then wants to switch to other `flag:N` term is replaced with correct flag, without removing existing search terms.
5. Only one flag can be selected at one time, when using Search by flag option.

## How Has This Been Tested?
* Added unit test for checking if my function is selecting correct cards.
* Tested it on emulated device in Android Studio (Pixel 3a API 30)

## Learning (optional, can help others)
None

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code

### CODE REVIEW

File: AnkiDroid/src/main/res/menu/card_browser.xml [hunk 1]
```diff
@@ -1,11 +1,40 @@
 <menu xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:ankidroid="http://schemas.android.com/apk/res-auto" >
 
+    <item
+        ankidroid:showAsAction="ifRoom"
```
Comment (david-allison-1) [inline AnkiDroid/src/main/res/menu/card_browser.xml:5]: I feel it'd be best to move this into the menu with the other search items (both removing it as an action, and repositioning it).

There's little visual feedback that the flag icon means "search", and it feels peculiar to give it priority over other methods of filtering.

```suggestion
        ankidroid:showAsAction="never"
```

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
Clean code, no bugs, works as expected, and there's testing! Thanks so much

One slight nitpick regarding the positioning of the menu item, but aside from that, good to go!

**Comment by Volfee** [comment]
@david-allison-1 I have applied the changes that you requested and cleaned code from unnecessary functions.

**Review by david-allison-1 — APPROVED** [review]
Looks good to me! Thanks!

**Comment by david-allison-1** [comment]
Took on a conflict due to 5fc915d8d02e6e3d4333819e98163e3e31b0e7d5. Sorry! 

`id`s are no longer final, so a change from a `switch` to an `if/else` is required

**Review by mikehardy — APPROVED** [review]
...yeah that was my fault - working in the same area :-), this was going to be a squash merge anyway so I attempted to resolve the conflicts.

I believe I did it all correctly, it looks good to me so I'm going to mark this pending merge. Thanks @Volfee !

**Pull request merged**